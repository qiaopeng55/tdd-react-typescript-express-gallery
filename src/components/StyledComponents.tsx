import { createStyles, Theme } from "@material-ui/core/styles";
import { fade } from "@material-ui/core/styles/colorManipulator";
import { PaginationClassKey } from "material-ui-flat-pagination";
import styled from "styled-components";

export const CardContainer = styled.div`
  background: white;
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin-top: 15px;
  border: 1px solid #eeeeee;

  .image {
    display: flex;
    justify-content: center;
    align-items: center;
    margin: 10px;
    height: 200px;

    img {
      height: 100%;
    }
  }

  .text__area {
    margin: 15px;
  }
`;

export const PaginationSetting = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

export const paginationStyles = (theme: Theme) =>
  createStyles<
    | "paperRoot"
    | Extract<
        PaginationClassKey,
        "text" | "colorInheritCurrent" | "colorInheritOther"
      >
  >({
    paperRoot: {
      margin: theme.spacing.unit * 2,
      padding: theme.spacing.unit * 2
    },
    text: {
      textTransform: "initial"
    },
    colorInheritCurrent: {
      color: "#000000",
      "&:hover": {
        // backgroundColor: "#ffffff",
        // borderBottom: "5px solid black",
        // borderRadius: "0px"
      },
      background: "#ffffff",
      borderBottom: "4px solid black",
      marginBottom: "-4px",
      borderRadius: "0px"
    },
    colorInheritOther: {
      color: "#223344",
      "&:hover": {
        backgroundColor: fade("#223344", theme.palette.action.hoverOpacity)
      }
    }
  });

export const WhiteSpace = styled.div`
  height: ${({ height = 10 }: { height?: number }) => height}px;
`;

export const Divider = styled.div`
  border: ${({ weight = 0.5 }: { weight?: number }) => weight}px solid #eeeeee;
`;
