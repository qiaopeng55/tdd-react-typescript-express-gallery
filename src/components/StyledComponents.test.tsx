import { shallow } from "enzyme";
import * as React from "react";
import { Divider, WhiteSpace } from "./StyledComponents";

describe("WhiteSpace", () => {
  const wrapper = shallow(<WhiteSpace />);

  it("renders properly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("Divider", () => {
  const wrapper = shallow(<Divider />);

  it("renders properly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
