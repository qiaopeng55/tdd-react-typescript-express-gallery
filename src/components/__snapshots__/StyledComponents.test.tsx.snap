// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForwardRef />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "forwardedClass": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bxivhb",
          "isStatic": false,
          "rules": Array [
            "
  border: ",
            [Function],
            "px solid #eeeeee;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bxivhb",
        "target": "div",
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
      "forwardedRef": null,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "forwardedClass": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-bxivhb",
            "isStatic": false,
            "rules": Array [
              "
  border: ",
              [Function],
              "px solid #eeeeee;
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-bxivhb",
          "target": "div",
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
        "forwardedRef": null,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`WhiteSpace renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForwardRef />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "forwardedClass": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": false,
          "rules": Array [
            "
  height: ",
            [Function],
            "px;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": "div",
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
      "forwardedRef": null,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "forwardedClass": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": false,
            "rules": Array [
              "
  height: ",
              [Function],
              "px;
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "div",
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
        "forwardedRef": null,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
